dspace-api/src/main/java/org/dspace/statistics/export/service/OpenUrlServiceImpl.java

```diff
@@ -0,0 +1,135 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.statistics.export.service;
+
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.net.URLConnection;
+import java.sql.SQLException;
+import java.util.Date;
+import java.util.List;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.log4j.Logger;
+import org.dspace.core.Context;
+import org.dspace.statistics.export.OpenURLTracker;
+import org.springframework.beans.factory.annotation.Autowired;
+
+/**
+ * Implementation of the OpenUrlService interface
+ */
+public class OpenUrlServiceImpl implements OpenUrlService {
+
+    private Logger log = Logger.getLogger(OpenUrlService.class);
+
+    @Autowired
+    protected FailedOpenURLTrackerService failedOpenUrlTrackerService;
+
+    /**
+     * Processes the url
+     * When the contacting the url fails, the url will be logged in a db table
+     * @param c - the context
+     * @param urlStr - the url to be processed
+     * @throws SQLException
+     */
+    public void processUrl(Context c, String urlStr) throws SQLException {
+        log.debug("Prepared to send url to tracker URL: " + urlStr);
+
+        try {
+            int responseCode = getResponseCodeFromUrl(urlStr);
+            if (responseCode != HttpURLConnection.HTTP_OK) {
+                logfailed(c, urlStr);
+            } else if (log.isDebugEnabled()) {
+                log.debug("Successfully posted " + urlStr + " on " + new Date());
+            }
+        } catch (Exception e) {
+            log.error("Failed to send url to tracker URL: " + urlStr);
+            logfailed(c, urlStr);
+        }
+    }
+
+    /**
+     * Returns the response code from accessing the url
+     * @param urlStr
+     * @return response code from the url
+     * @throws IOException
+     */
+    protected int getResponseCodeFromUrl(final String urlStr) throws IOException {
+        URLConnection conn;
+        URL url = new URL(urlStr);
+        conn = url.openConnection();
+
+        return ((HttpURLConnection) conn).getResponseCode();
```

Comment:
This Connection never seems to be closed? I think we need to close the connection here after getting the responseCode.